contracts/zero-ex/src/rollback.ts

```diff
@@ -0,0 +1,339 @@
+                            { title: selector, value: selector, description: signature },
+                            { title: signature, value: selector, description: selector },
+                        ]),
+                        ...proxyFunctions
+                            .filter(fn => !Object.keys(selectorToSignature).includes(fn.id))
+                            .map(fn => ({ title: fn.id, value: fn.id, description: '(function signature not found)' })),
+                    ],
+                });
+                const functionEntity = proxyFunctions.find(fn => fn.id === fnSelector);
+                if (functionEntity === undefined) {
+                    logUtils.log(`Couldn't find deployment history for selector ${fnSelector}`);
+                } else {
+                    logUtils.log(
+                        functionEntity.fullHistory.map(update => ({
+                            date: timestampToUTC(update.timestamp),
+                            impl: update.impl,
+                        })),
+                    );
+                }
+                break;
+            case CommandLineActions.Current:
+                const currentFunctions: {
+                    [selector: string]: { signature: string; impl: string; lastUpdated: string };
+                } = {};
+                proxyFunctions
+                    .filter(fn => fn.currentImpl !== constants.NULL_ADDRESS)
+                    .map(fn => {
+                        currentFunctions[fn.id] = {
+                            signature: selectorToSignature[fn.id] || '(function signature not found)',
+                            impl: fn.currentImpl,
+                            lastUpdated: timestampToUTC(fn.fullHistory.slice(-1)[0].timestamp),
+                        };
+                    });
+                logUtils.log(currentFunctions);
+                break;
+            case CommandLineActions.Rollback:
+                if (provider === undefined) {
+                    const { rpcUrl } = await prompts({
+                        type: 'text',
+                        name: 'rpcUrl',
+                        message: 'Enter an RPC endpoint:',
+                    });
+                    provider = createWeb3Provider(rpcUrl);
+                }
+                const chainId = await new Web3Wrapper(provider).getChainIdAsync();
+                const { exchangeProxy } = getContractAddressesForChainOrThrow(chainId);
+                const zeroEx = new wrappers.IZeroExContract(exchangeProxy, provider);
+
+                const { selected } = await prompts({
+                    type: 'autocompleteMultiselect',
+                    name: 'selected',
+                    message: 'Select the functions to rollback:',
+                    choices: _.flatMap(
+                        proxyFunctions.filter(
+                            fn => fn.currentImpl !== constants.NULL_ADDRESS && !DO_NOT_ROLLBACK.includes(fn.id),
```

Comment:
Since we're picking the functions manually here anyway, do we need the blacklist? E.g., I could see us upgrading `migrate()` and then changing our mind.

Reply:
yeah I guess that's true